CMakeLists.txt

```diff
@@ -806,12 +805,44 @@ endif(WITH_LIBJANSSON STREQUAL "BUNDLED")
 list(APPEND EP_INCLUDES ${LIBJANSSON_INCLUDES})
 list(APPEND EP_LIBS ${LIBJANSSON_LIBS})
 
+# WITH_LIBNCURSES can have multiple values with different meanings
+# on Linux:
+# * "SYSTEM" - (default) uses ncurses installed in the system
+# * "GIT"    - does a git clone and builds it
+if(UNIX)
+  if(WITH_LIBNCURSES STREQUAL "GIT")
+    set(LIBNCURSES_TARGET libncurses)
+    externalproject_add(${LIBNCURSES_TARGET}
+      GIT_REPOSITORY    https://github.com/mirror/ncurses.git
+      GIT_TAG           v6.0
+      UPDATE_COMMAND    ""
+      CONFIGURE_COMMAND ${DEFAULT_CONFIGURE_OPTS} "CFLAGS=${CFLAGS} -fPIC" "CXXFLAGS=${CXXFLAGS} -fPIC" --with-normal --without-debug --without-ada --with-cxx-binding --with-termlib=tinfo --with-ticlib=tic
+      BUILD_COMMAND     make all AUTOCONF=: AUTOHEADER=: AUTOMAKE=: ACLOCAL=:
+      INSTALL_COMMAND   make install AUTOCONF=: AUTOHEADER=: AUTOMAKE=: ACLOCAL=:
+      )
+    set(LIBNCURSES ${CMAKE_CURRENT_BINARY_DIR}/external/lib/libncurses.a)
+    set(LIBTINFO ${CMAKE_CURRENT_BINARY_DIR}/external/lib/libtinfo.a)
+
+    # set curses variables, for more info check https://cmake.org/cmake/help/v3.10/module/FindCurses.html
+    set(CURSES_FOUND TRUE)
+    set(CURSES_LIBRARIES ${LIBNCURSES};${LIBTINFO})
+    set(CURSES_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/external/include/ncurses)
+    set(CURSES_HAVE_CURSES_H ${CURSES_INCLUDE_DIR}/curses.h)
+    set(CURSES_HAVE_NCURSES_H ${CURSES_INCLUDE_DIR}/ncurses.h)
+
+    list(APPEND EP_TARGETS ${LIBNCURSES_TARGET})
+  elseif(WITH_LIBNCURSES STREQUAL "SYSTEM")
+    include(FindCurses)
+    find_package(Curses REQUIRED)
+  endif()
+endif(UNIX)
+
 if(WITH_LIBEDIT STREQUAL "BUNDLED")
```

Comment:
The default WITH_LIBEDIT option is no longer "BUNDLED". I don't think it is necessary to change it now, but will have to be updated with CBRD-21611.

Reply:
Yes, you are right I'll cover this in [CBRD-21611](http://jira.cubrid.org/browse/CBRD-21611)